make/test.mk

```diff
@@ -163,7 +166,14 @@ execute-tests:
 	@echo "Status of ToolchainStatus"
 	-oc get ToolchainStatus -n ${HOST_NS} -o yaml
 	@echo "Starting test $(shell date)"
-	MEMBER_NS=${MEMBER_NS} MEMBER_NS_2=${MEMBER_NS_2} HOST_NS=${HOST_NS} REGISTRATION_SERVICE_NS=${REGISTRATION_SERVICE_NS} go test ${TESTS_TO_EXECUTE} -run ${TESTS_RUN_FILTER_REGEXP} -p 1 -parallel ${E2E_PARALLELISM} -v -timeout=90m -failfast || \
+ifneq ($(OPENSHIFT_BUILD_NAMESPACE),) 
+	$(MAKE) check-go-junit-report
+else 
+	@echo "Skipping Go Junit report check and install"
+endif
+	@echo "Running tests"
+	MEMBER_NS=${MEMBER_NS} MEMBER_NS_2=${MEMBER_NS_2} HOST_NS=${HOST_NS} REGISTRATION_SERVICE_NS=${REGISTRATION_SERVICE_NS} go test ${TESTS_TO_EXECUTE} -run ${TESTS_RUN_FILTER_REGEXP} -p 1 -parallel ${E2E_PARALLELISM} -v -timeout=90m -failfast 2>&1 | tee test_output.log
+	$(MAKE) generate-report REPORT_NAME=${REPORT_NAME} || \
 	($(MAKE) print-logs HOST_NS=${HOST_NS} MEMBER_NS=${MEMBER_NS} MEMBER_NS_2=${MEMBER_NS_2} REGISTRATION_SERVICE_NS=${REGISTRATION_SERVICE_NS} && exit 1)
```

Comment:
this is executed as a separate command, which means that printing logs will depend on the result of the `generate-report` target, not on the result of the tests. 
The original way it was written printed out the logs only when the test failed and also returned the exit code `1`, so execution of the `print-logs` target depended on the results of the tests - this needs to stay the same.

Reply:
so i am now running the generate report cmd after print logs hope this works